consensus/consensus-types/src/sync_info.rs

```diff
@@ -95,16 +102,16 @@ impl SyncInfo {
 
         ensure!(
             self.highest_quorum_cert.certified_block().round()
-                >= self.highest_commit_cert.certified_block().round(),
+                >= self.highest_commit_cert().certified_block().round(),
             "HQC has lower round than HCC"
         );
         ensure!(
-            *self.highest_commit_cert.commit_info() != BlockInfo::empty(),
+            *self.highest_commit_cert().commit_info() != BlockInfo::empty(),
             "HCC has no committed block"
         );
         self.highest_quorum_cert
             .verify(validator)
-            .and_then(|_| self.highest_commit_cert.verify(validator))
+            .and_then(|_| self.highest_commit_cert().verify(validator))
```

Comment:
this is where we can save the verification cost. 
`self.highest_commit_cert.map_or(Ok(()), |cert| cert.verify(validator)`